packages/block-library/src/cover/edit.native.js

```diff
@@ -0,0 +1,221 @@
+/**
+ * External dependencies
+ */
+import { View } from 'react-native';
+
+/**
+ * WordPress dependencies
+ */
+import { __ } from '@wordpress/i18n';
+import {
+	Icon,
+	ImageWithFocalPoint,
+	PanelBody,
+	RangeControl,
+} from '@wordpress/components';
+import {
+	InnerBlocks,
+	InspectorControls,
+	MEDIA_TYPE_IMAGE,
+	MediaPlaceholder,
+	withColors,
+} from '@wordpress/block-editor';
+import { compose, withPreferredColorScheme } from '@wordpress/compose';
+import { withSelect } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { cover as icon } from '@wordpress/icons';
+
+/**
+ * Internal dependencies
+ */
+import styles from './style.scss';
+import { onCoverSelectMedia } from './edit.js';
+import { COVER_MIN_HEIGHT, IMAGE_BACKGROUND_TYPE, VIDEO_BACKGROUND_TYPE } from './shared';
+
+/**
+ * Constants
+ */
+const ALLOWED_MEDIA_TYPES = [ MEDIA_TYPE_IMAGE ];
+const INNER_BLOCKS_TEMPLATE = [
+	[ 'core/paragraph', {
+		align: 'center',
+		fontSize: 'large',
+		placeholder: __( 'Write title…' ),
+	} ],
+];
+const COVER_MAX_HEIGHT = 1000;
+const COVER_DEFAULT_HEIGHT = 300;
+
+const Cover = ( {
+	attributes,
+	isAncestorSelected,
+	isParentSelected,
+	isSelected,
+	onFocus,
+	overlayColor,
+	setAttributes,
+} ) => {
+	const {
+		backgroundType,
+		dimRatio,
+		focalPoint,
+		gradientValue,
+		minHeight,
+		url,
+	} = attributes;
+	const [ containerSize, setContainerSize ] = useState( null );
+	const CONTAINER_HEIGHT = minHeight || COVER_DEFAULT_HEIGHT;
+
+	// Used to set a default color for its InnerBlocks
+	// since there's no system to inherit styles yet
+	// the RichText component will check if there are
+	// parent styles for the current block. If there are,
+	// it will use that color instead.
+	useEffect( () => {
+		setAttributes( { childrenStyles: styles.defaultColor } );
+	}, [ setAttributes ] );
+
+	const onSelectMedia = ( media ) => {
+		const onSelect = onCoverSelectMedia( setAttributes );
+		onSelect( media );
+	};
+
+	const onHeightChange = ( value ) => {
+		if ( minHeight || value !== COVER_DEFAULT_HEIGHT ) {
+			setAttributes( { minHeight: value } );
+		}
+	};
+
+	const onOpactiyChange = ( value ) => {
+		setAttributes( { dimRatio: value } );
+	};
+
+	const onContainerLayout = ( event ) => {
+		const { height, width } = event.nativeEvent.layout;
+		setContainerSize( { width, height } );
+	};
+
+	const getOpacity = () => {
+		// Set opacity to 1 while video support is not available
+		if ( VIDEO_BACKGROUND_TYPE === backgroundType ) {
+			return 1;
+		}
+
+		if ( url ) {
+			return dimRatio !== 0 ? dimRatio / 100 : 0;
```

Comment:
Interesting, how is this different than just doing `return dimRatio / 100`?

Reply:
I seriously can't recall why I did this 😆 but yeah no need for that, updated it here https://github.com/WordPress/gutenberg/pull/19722/commits/e06c903806a50e32f1adc5d6dffeca00bb13aea9 (sorry this commit has more changes due to prettier formatting)